DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,400 @@
+            {
+                return updateList;
+            }
+            else if (updateList == null)
+            {
+                return null;
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+                List<DiscordApplicationCommand> NewCommands = new();
+                foreach (var cmd in updateList)
+                {
+                    if (discord == null || !discord.Any(d => d.Name == cmd.Name))
+                    {
+                        NewCommands.Add(cmd);
+                    }
+                }
+
+                return NewCommands;
+            }
+        }
+
+        private static Dictionary<ulong, DiscordApplicationCommand> BuildGuildOverwriteList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any() || !ApplicationCommandsExtension._guildDiscordCommands.Any(l => l.Key == guildId))
+            {
+                return null;
+            }
+            else if (updateList == null)
+            {
+                return null;
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+                if (discord != null)
+                {
+                    Dictionary<ulong, DiscordApplicationCommand> UpdateCommands = new();
+                    foreach (var cmd in updateList)
+                    {
+                        if (discord.Any(d => d.Name == cmd.Name))
+                        {
+                            UpdateCommands.Add(discord.Where(d => d.Name == cmd.Name).First().Id, cmd);
+                        }
+                    }
+                    return UpdateCommands;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        private static List<ulong> BuildGlobalDeleteList(List<DiscordApplicationCommand> updateList = null)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any())
+            {
+                return null;
+            }
+            else if (updateList == null)
+            {
+                var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
```

Comment:
Remove surrounding else block here. It's assumed that reaching this point meant the above statement was false